test/runtests.jl

```diff
@@ -86,3 +86,73 @@ end
   @test nls_model.nls_meta.nequ == 256 * 256
   @test all(0 .<= nls_model.meta.x0 .<= 1)
 end
+
+@testset "SVM-Train" begin
+  nlp_train,nls_train,sol = svm_train_model()
+  @test typeof(nlp_train) <: FirstOrderModel
+  @test typeof(nls_train) <: FirstOrderNLSModel
+  @test typeof(sol) == Vector{Int64}
+
+
+  x = nlp_train.meta.x0
+  f = obj(nlp_train, x)
+  F = residual(nls_train, x)
+  @test f ≈ dot(F, F) / 2
+
+  g = grad(nlp_train, x)
+  JtF = jtprod_residual(nls_train, x, F)
+  @test all(g .≈ JtF)
+  ### below is TBD
+  # @test model.meta.nvar == 5
+  # @test model.nls_meta.nequ == 202
+  # @test all(model.meta.x0 .== 1)
+  # @test length(findall(x -> x .!= 0, sol)) == 2
+end
+
+@testset "SVM-Test" begin
+  nlp_test,nls_test,sol = svm_test_model()
+  @test typeof(nlp_test) <: FirstOrderModel
+  @test typeof(nls_test) <: FirstOrderNLSModel
+  @test typeof(sol) == Vector{Int64}
+
+  x = nlp_test.meta.x0
+  f = obj(nlp_test, x)
+  F = residual(nls_test, x)
+  @test f ≈ dot(F, F) / 2
+
+  g = grad(nlp_test, x)
+  JtF = jtprod_residual(nls_test, x, F)
+  @test all(g .≈ JtF)
+  ### Below is TBD
+  # @test model.meta.nvar == 5
+  # @test model.nls_meta.nequ == 202
+  # @test all(model.meta.x0 .== 1)
+  # @test length(findall(x -> x .!= 0, sol)) == 2
+end
+
+##### Obsolete since svm_train/test_model() does not yield resid() anymore
+# function comp_derivs()
+#   nlp_train, nlp_test, sol_train = svm_train_model() #
+# nls_train = ADNLSModel(resid_train, ones(size(nls_train.meta.x0)),size(sol_train,1) + size(nls_train.meta.x0,1))
+# f = ReverseADNLSModel(resid!, size(sol_train,1), ones(size(model_train.meta.x0)), name = "Dominique")
+#   fad = ReverseADNLSModel(resid!, size(sol_train,1), ones(size(model_train.meta.x0)), name = "Dominique")
+
+#   xk = 10*randn(size(fk.meta.x0));
+#   v = 10*randn(size(xk));
+#   Jvac = zeros(size(sol_train));
+#   Jvac_ad = similar(Jvac);
+#   @show @benchmark jprod_residual!($fk, $xk, $v, $Jvac)
+#   @show @benchmark jprod_residual!($fad, $xk, $v, $Jvac_ad)
+
+#   # @show norm(Jvac - Jvac_ad)
+
+#   v = 10*randn(size(sol_train));
+#   Jtvac = zero(xk);
+#   Jtvac_ad = zero(xk);
+
+#   @show @benchmark jtprod_residual!($fk, $xk, $v, $Jtvac)
+#   @show @benchmark jtprod_residual!($fad, $xk, $v, $Jtvac_ad)
+
+#   @show norm(Jtvac - Jtvac_ad)
```

Comment:
`resid()` is in the NLSModel as `nls_train.residual()`. Any additional tests are super useful here.

Reply:
Agreed but this requires ReverseADNLSModels, which isn't registered yet. What do you want to do?

Reply:
Oh I see. Sorry, nevermind.